### PR TITLE
188527686 formula editor improvements

### PR DESCRIPTION
**Pull request by eireland**
Makes formula editor modal resizable.
Removes default codap modal header.

Deployed here:
https://codap3.concord.org/branch/formula-editor-improvements/?sample=coasters

### CODE REVIEW

**Comment by bfinzer** [comment]
I'm seeing two lint warnings: `ModalCloseButton` and `ModalHeader` are defined but not used in edit-formula-modal.tsx.

**Comment by bfinzer** [comment]
In the V2 formula editor text wraps when it reaches the right edge. This should be the behavior in V3 as well. If this is easy to bring about, it could be part of this PR, otherwise a separate story.

**Review by bfinzer — APPROVED** [review]
👍🏻LGTM
Just a couple notes.

File: v3/src/components/common/edit-formula-modal.tsx [hunk 1]
```diff
@@ -28,10 +29,14 @@ interface IProps {
 export const EditFormulaModal = observer(function EditFormulaModal({
   applyFormula, formulaPrompt, isOpen, onClose, titleInput, titleLabel, titlePlaceholder, value
 }: IProps) {
+  const minWidth = 400
+  const minHeight = 180
   const [showValuesMenu, setShowValuesMenu] = useState(false)
   const [showFunctionMenu, setShowFunctionMenu] = useState(false)
   const formulaEditorState = useFormulaEditorState(value ?? "")
   const { formula, setFormula } = formulaEditorState
+  const [dimension, setDimension] = useState({ width: minWidth, height: minHeight })
```
Comment (kswenson) [inline v3/src/components/common/edit-formula-modal.tsx:38]:
```suggestion
  const [dimensions, setDimensions] = useState({ width: minWidth, height: minHeight })
```

File: v3/src/components/common/edit-formula-modal.tsx [hunk 2]
```diff
@@ -47,6 +52,7 @@ export const EditFormulaModal = observer(function EditFormulaModal({
     setShowFunctionMenu(false)
     setFormula(value || "")
     onClose?.()
+    setDimension({ width: minWidth, height: minHeight })
```
Comment (kswenson) [inline v3/src/components/common/edit-formula-modal.tsx:55]: Are we sure we want to reset the size on close? I can see an argument that retaining its size across invocation would be desirable.

File: v3/src/components/common/edit-formula-modal.tsx [hunk 3]
```diff
@@ -28,10 +29,14 @@ interface IProps {
 export const EditFormulaModal = observer(function EditFormulaModal({
   applyFormula, formulaPrompt, isOpen, onClose, titleInput, titleLabel, titlePlaceholder, value
 }: IProps) {
+  const minWidth = 400
+  const minHeight = 180
```
Comment (kswenson) [inline v3/src/components/common/edit-formula-modal.tsx:33]: These values appear here and in the CSS. Either import the values from the .css or use CSS variables, e.g. `--default-codap-modal-width` and `--default-codap-modal-height`, which are accessible from CSS and JS.

File: v3/src/components/common/edit-formula-modal.tsx [hunk 4]
```diff
@@ -86,20 +92,44 @@ export const EditFormulaModal = observer(function EditFormulaModal({
     event.stopPropagation()
   }
 
+  const handleResizeModal = useCallback((e: React.PointerEvent) => {
+    if (e.pointerId !== undefined) {
+      e.currentTarget.setPointerCapture(e.pointerId)
+    }
+
+    const modalRect = document.querySelector(".codap-modal-content")?.getBoundingClientRect()
```
Comment (kswenson) [inline v3/src/components/common/edit-formula-modal.tsx:100]: Document-wide queries are fragile -- what if there are two modals showing? Either use a ref or a library like `react-resize-detector`.

File: v3/src/components/common/edit-formula-modal.tsx [hunk 5]
```diff
@@ -86,20 +92,44 @@ export const EditFormulaModal = observer(function EditFormulaModal({
...
+    let resizingWidth = startWidth, resizingHeight = startHeight
+
+    const onPointerMove = (pointerMoveEvent: { pageX: number; pageY: number }) => {
+      const xDelta = pointerMoveEvent.pageX - startPosition.x
+      const yDelta = pointerMoveEvent.pageY - startPosition.y
+      resizingWidth = Math.max(startWidth + xDelta, minWidth)
+      resizingHeight = Math.max(startHeight + yDelta, minHeight)
+      setDimension({
+        width: resizingWidth, height: resizingHeight,
+      })
```
Comment (kswenson) [inline v3/src/components/common/edit-formula-modal.tsx:114]: I recommend rounding these values to integer pixels -- they're much easier to deal with when debugging and I don't think fractional pixel widths are necessary for users.

File: v3/src/components/common/edit-formula-modal.scss [hunk 6]
```diff
@@ -1,10 +1,23 @@
 @use "../vars";
 
 .codap-modal-content {
+  min-height: 180px;
+  min-width: 400px;
```
Comment (kswenson) [inline v3/src/components/common/edit-formula-modal.scss:5]: These values appear here and in the JS. They should be shared. See comment elsewhere for more details.

**Review by kswenson — CHANGES_REQUESTED** [review]
👍 Generally looks good -- see inline suggestions.

>In the V2 formula editor text wraps when it reaches the right edge. This should be the behavior in V3 as well. If this is easy to bring about, it could be part of this PR, otherwise a separate story.

According to ChatGPT, CodeMirror can be configured to wrap text by adding `EditorView.lineWrapping` to the list of extensions when initializing the editor.

File: v3/src/components/common/edit-formula-modal.scss [hunk 7]
```diff
@@ -1,10 +1,33 @@
...
+  min-width: $edit-formula-modal-min-width-px;
+
   .formula-modal-body {
     cursor: default;
+    display: flex;
+    flex-direction: column;
+    height: inherit;
+    margin-bottom: 10px;
+    min-height: 72px;
+    min-width: 400px;
```
Comment (kswenson) [inline v3/src/components/common/edit-formula-modal.scss:24]: Could this be:
```suggestion
    min-width: $edit-formula-modal-min-width-px;
```

File: v3/src/components/common/edit-formula-modal.scss [hunk 8]
```diff
@@ -13,24 +36,63 @@
...
-    .formula-editor-button {
-      padding: 0 10px;
-      &:hover {
-        background: vars.$codap-teal-lighter;
-      }
-      &:active, &.menu-open {
-        background: vars.$codap-teal-light;
-        color: white;
+        .formula-editor-input {
+          width: calc(100% - 50px);
```
Comment (kswenson) [inline v3/src/components/common/edit-formula-modal.scss:45]: Comment or named constant for the `50px` value

File: v3/src/components/common/edit-formula-modal.scss [hunk 9]
```diff
@@ -13,24 +36,63 @@
...
     cursor: default;
+    border-radius: 6px;
+    position: relative;
+    bottom: 0;
+    right: 0;
+  }
+
+  .codap-modal-corner {
+    position: absolute;
+    bottom: 0px;
```
Comment (kswenson) [inline v3/src/components/common/edit-formula-modal.scss:86]:
```suggestion
    bottom: 0;
```
For consistency

**Review by kswenson — APPROVED** [review]
👍 Looks good -- just a few small suggestions.

**Pull request merged**